### ISSUE TITLE
 Inquiry about multi-image input and dialogue support in InstructBLIP

### ISSUE THREAD

**Issue by pygh0er** [salesforce/LAVIS#318]
Hi,

Based on my examination of the code, it appears that InstructBLIP concatenates the query of the images directly before the textual input embeddings. This leads me to believe that multi-image input and dialogue support may not be currently available in the codebase.

Could you kindly confirm whether my understanding is correct? Furthermore, I am curious to know if there are any plans to introduce support for multi-image input and dialogue in the future roadmap of InstructBLIP.:smile:

`inputs_embeds = self.llm_model.get_input_embeddings()(llm_tokens.input_ids)`
 `inputs_embeds = torch.cat([inputs_llm, inputs_embeds], dim=1)  `
 `attention_mask = torch.cat([atts_llm, llm_tokens.attention_mask], dim=1)`

**Comment by LiJunnan1992** [comment]
For multi-image, you can concatenate all images' query embeddings (i.e. output from qformer) before the text embedding. 
For dialog, you can consider the dialog history as part of the text.

**Comment by pygh0er** [comment]
> For multi-image, you can concatenate all images' query embeddings (i.e. output from qformer) before the text embedding. For dialog, you can consider the dialog history as part of the text.

Thans for your response! :)

I have been considering the scenario with multiple images where the position of the images in the text is crucial for the semantics. Actually, I have some concerns that concatenating the representations of multiple images at the beginning of the text may not yield satisfactory results.  But I will validate your suggestion in my task immediately.

I sincerely appreciate your feedback!

**Comment by lxr-1204** [comment]
Have you verified the effect of multi-image input? Can you tell me whether the result is satisfactory?
